starknet_py/utils/typed_data.py

```diff
@@ -169,6 +134,45 @@ def to_dict(self) -> dict:
     def _is_struct(self, type_name: str) -> bool:
         return type_name in self.types
 
+    def _encode_value_v1(
+        self, basic_type: BasicType, value: Union[int, str, dict, list]
+    ) -> Optional[int]:
+        if basic_type in (
+            BasicType.FELT,
+            BasicType.SHORT_STRING,
+            BasicType.CONTRACT_ADDRESS,
+            BasicType.CLASS_HASH,
+        ) and isinstance(value, (int, str)):
+            return parse_felt(value)
+
+        if basic_type in (
+            BasicType.U128,
+            BasicType.TIMESTAMP,
+        ) and isinstance(value, (int, str)):
+            return encode_u128(value)
+
+        if basic_type == BasicType.I128 and isinstance(value, (int, str)):
+            return encode_i128(value)
+
+        if basic_type == BasicType.STRING and isinstance(value, str):
+            return self._prepare_long_string(value)
+
+        return None
+
+    # pylint: disable=no-self-use
+    def _encode_value_v0(
```

Comment:
`_encode_value_v0()` could be a function, however having it and `_encode_value_v1()` as respectively function and `TypedData` method doesn't look good. Let's keep both as methods.